forc-util/src/fs_locking.rs

```diff
@@ -201,4 +236,32 @@ mod test {
         let e = metadata(&x.0).unwrap_err().kind();
         assert_eq!(e, ErrorKind::NotFound);
     }
+
+    #[test]
+    fn test_cleanup_stale_files() {
+        // First create some test files
+        let test_lock = PidFileLocking::lsp("test_cleanup");
+        test_lock.lock().expect("Failed to create test lock file");
+
+        // Create a test lock file with invalid PID
+        let lock_path = user_forc_directory()
+            .join(".lsp-locks")
+            .join("test_cleanup.lock");
+        File::create(&lock_path).expect("Failed to create test lock file");
+
+        // Run cleanup and check returned paths
+        let cleaned_paths =
+            PidFileLocking::cleanup_stale_files().expect("Failed to cleanup stale files");
+
+        // Verify that only the invalid lock file was cleaned up
+        assert_eq!(cleaned_paths.len(), 1);
+        assert_eq!(cleaned_paths[0], lock_path);
+
+        // Verify file system state
+        assert!(test_lock.0.exists(), "Active lock file should still exist");
+        assert!(!lock_path.exists(), "Lock file should be removed");
+
+        // Cleanup after test
+        test_lock.release().expect("Failed to release test lock");
```

Comment:
nit: not a biggie, maybe we should use tempfile to ensure file/dir cleanup (automatically done when var goes out of scope - even if a test fails mid-way)